code/modules/medical/genetics/bioEffects/harmful.dm

```diff
@@ -707,31 +707,40 @@
 	stability_loss = -10
 	var/const/radius = 2
 	icon_state  = "bad"
+	var/roundedmultremainder
 
-	OnLife()
+	OnLife(var/mult)
 		..()
-		if (prob(50))
-			return
-
+		var/roundedmult = round(mult)
+		roundedmultremainder += (mult % 1)
+		if (roundedmultremainder >= 1)
+			roundedmult += round(roundedmultremainder)
+			roundedmultremainder = roundedmultremainder % 1
 		var/turf/T
-		//don't really need this but to make it more harmful to the user.
-		if (prob(5))
-			T = get_turf(owner)
-		else
-			T = locate(owner.x + rand(-radius/2,radius+2), owner.y+rand(-radius/2,radius/2), 1)
-
-		var/obj/overlay/pulse = new/obj/overlay(T)
-		pulse.icon = 'icons/effects/effects.dmi'
-		pulse.icon_state = "emppulse"
-		pulse.name = "emp pulse"
-		pulse.anchored = 1
-		SPAWN_DBG(2 SECONDS)
-			if (pulse) qdel(pulse)
-
-		//maybe have this only emp some things on the tile.
-		if(istype(T))
-			for (var/atom/O in T.contents)
-				O.emp_act()
+		for (roundedmult = roundedmult, roundedmult > 0, roundedmult --)
+			if (prob(50))
+				if (prob(5)) //don't really need this but to make it more harmful to the user.
+					T = get_turf(owner)
+				else
+					T = locate(owner.x + rand(-radius/2,radius+2), owner.y+rand(-radius/2,radius/2), owner.z)
```

Comment:
Since this is location dependent it does seem a little goofy that you could then get a bunch of emps that burst when server gets behind for a while. 🤷